src/components/Cluster/ClusterDetail/AddNodePool/AddNodePoolAZSelectionCheckbox.tsx

```diff
@@ -0,0 +1,58 @@
+import { AvailabilityZoneSelection } from 'Cluster/ClusterDetail/AddNodePool/AddNodePoolUtils';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import RUMActionTarget from 'RUM/RUMActionTarget';
+import { RUMActions } from 'shared/constants/realUserMonitoring';
+import RadioInput from 'UI/Inputs/RadioInput';
+import { mergeActionNames } from 'utils/realUserMonitoringUtils';
+
+interface IAddNodePoolAZSelectionCheckboxProps
+  extends Omit<
+    React.ComponentPropsWithoutRef<typeof RadioInput>,
+    'onChange' | 'value' | 'id'
+  > {
+  onChange: (newAZSelection: AvailabilityZoneSelection) => void;
+  type?: AvailabilityZoneSelection;
+  value?: AvailabilityZoneSelection;
+  npID?: string;
+}
+
+const AddNodePoolAZSelectionCheckbox: React.FC<IAddNodePoolAZSelectionCheckboxProps> = ({
+  onChange,
+  type,
+  value,
+  npID,
+  ...rest
+}) => {
+  const typeName = AvailabilityZoneSelection[type as AvailabilityZoneSelection];
+  const id = `np-${npID}-az-${typeName.toLowerCase()}`;
+
+  return (
+    <RUMActionTarget
+      name={mergeActionNames(RUMActions.SelectAZSelection, typeName)}
```

Comment:
The RUM action name is now computed here, by appending the selection type to the end of the base action name (e.g. `SELECT_AZ_SELECTION_AUTOMATIC`)